### ISSUE TITLE
VS 2019 16.9.0 Preview 4.0 Issues

### ISSUE THREAD

**Issue by gmck** [xamarin/xamarin-android#5609]
VS hangs during the build (not responding). The build eventually completes (about normal timing) and the output window is then fully populated. Nothing appears in the Building Logging window until the build is finished. Attaching build output.txt and the binlog.

@dellis1972 Previous problems with not being able to deploy to the device from an aab seem to be fixed.

Issue #4317 appears to because my project has a nav_graph.xml in Resources.navigation. If I temporally exclude the navigation graph from the project I can open any layout file. If it is included then all layout files open with the error noted in #4317.

Can you advise if any of the following are still necessary in the top-level PropertyGroup?
```
<Deterministic>True</Deterministic>
<_AndroidAllowDeltaInstall>false</_AndroidAllowDeltaInstall>
```
 There is no documentation available at https://docs.microsoft.com/en-au/xamarin/android/release-notes which is what I usually rely on to determine if a preview is worth downloading. Last entry is November 2020 Xamarin.Android 11.1.99.168 in Visual Studio 2019 16.9 Preview 1. We are now up to 11.2.0.21 with Prev 4.0

[buildoutput.txt](https://github.com/xamarin/xamarin-android/files/5955495/buildoutput.txt)

[OBDNowPro_Debug_AnyCPU_Build_2021-02-10T11_48_45.9177157+11_00.zip](https://github.com/xamarin/xamarin-android/files/5955523/OBDNowPro_Debug_AnyCPU_Build_2021-02-10T11_48_45.9177157%2B11_00.zip)

**Comment by dellis1972** [comment]
We defaulted `_AndroidAllowDeltaInstall` too false in the latest preview to you can remove that one. `Deterministic` is an MSbuild property, so we don't set that so you might want to keep that. 
Due to some team restructuring the release notes are a bit delayed, but we are working on getting those sorted. 
I can say that P4 does include a bunch of fast deployment fixes (allot based on your feedback , thank you :) ). 

For the  Designer issue, I'm afraid garuma has moved on from Microsoft so its possible that issue was overlooked.  I have pinged someone on the team.

Looking at the build log, D8 is taking 11 seconds to build. Which might be ok if this is a decent sized project. That *might* be what is causing the hang. I'm not sure though. 

@joj any ideas on the IDE lockup here? We have not changed anything in the Aapt2 areas where we could normally see those kinds of UI issues (and I thought we'd fixed that with the daemon).

**Comment by joj** [comment]
We had a hack to show build output before the build completed, but that was troubling for a number of reasons. If you want you can switch to a more verbose build (detailed or diagnostic) and you'll have output from the start. It may feel like a hang, VS not showing anything, but it's actually compiling and then dumping the whole thing once it's done.

**Comment by gmck** [comment]
@joj, @dellis1972. I've already got diagnostic set, plus it has `<_FastDeploymentDiagnosticLogging>true</_FastDeploymentDiagnosticLogging>` set from earlier requests from Dean.

There is a couple of lines of output at the start. Doesn't stop at the same place every time. 

Example stops at line 2316. Hangs for the next 39 seconds, while the build completes and the final line number reported was 104389. So you don't get all the output at the start. 

Final line states TimeElapsed 27.81secs.  So VS now can't even count properly!!

You've got to be kidding if you are suggesting that is acceptable. Many would just give up when the "no responding" first appears and immediately kill VS.

**Comment by gmck** [comment]
@dellis1972 , @joj 

I skipped the last couple of previews, but got back on track a couple of days ago. I still get the not responding in Version 16.10.0 Preview 1.0. 

Do you need any further logs?

**Comment by joj** [comment]
that's by design in Visual Studio. The log window is fairly slow (in terms of build performance, at least) so VS delays the showing of information there to avoid making the build slower. That's only for Normal or Minimal log verbosity, though. The workaround is to make the log detailed or diagnostic. Honestly, I think any log lower than detailed is not providing information at all, anyway. We can't put that hack back. VS (the core team) didn't like it much and they did let us know :)

**Comment by dellis1972** [comment]
@gmck you can turn `_FastDeploymentDiagnosticLogging` off from that other issue. It will spam a TON of data that you don't need which is why we usually have it turned off by default. So that probably isn't helping. Sorry I forgot to mention to turn it off after we sorted that other issue.

**Comment by gmck** [comment]
@dellis1972, @joj 
 
I have tried `_FastDeploymentDiagnosticLogging` both on and off, but it makes no difference regarding the problem of the VS hanging while building. I now have it false by default and can obviously turn it back on again if and when you request it. 

My point was the workaround doesn't work as @joj describes.

The following sums it up
Output verbosity  Normal - OK
Logfile verbosity   Normal - OK

Output verbosity  Detailed - Hangs
Logfile verbosity   Detailed - Hangs

Output verbosity  Detailed - Hangs
Logfile verbosity   Diagnostic - Hangs

Output verbosity  Normal - OK
Logfile verbosity   Detailed - OK

Output verbosity  Normal - OK
Logfile verbosity   Diagnostic - OK

So anything above Output Normal will hang VS.  

A fast machine helps!!. In the old days I used to count the pixels on my screen when RTLink was linking - all you guys are probably too young to remember that sort of boredom.
